herddb-core/src/main/java/herddb/core/DBManager.java

```diff
@@ -218,9 +220,19 @@ public Thread newThread(final Runnable r) {
         preparedStatementsCache = new ServerSidePreparedStatementCache(statementsMem);
         String plannerType = serverConfiguration.getString(ServerConfiguration.PROPERTY_PLANNER_TYPE,
                 ServerConfiguration.PROPERTY_PLANNER_TYPE_DEFAULT);
+        PlansCache plansCache = new PlansCache(planCacheMem);
         switch (plannerType) {
             case ServerConfiguration.PLANNER_TYPE_CALCITE:
-                planner = new CalcitePlanner(this, planCacheMem);
+                planner = new CalcitePlanner(this, plansCache);
+                break;
+            case ServerConfiguration.PLANNER_TYPE_JSQLPARSER:
```

Comment:
If a user creates some tables with calcite configuration and then switches back to using plain jsql parser will there be any issues? Would there be any issues?

Reply:
No problem.
It is not about data, but only about how we access it.
Most probably it will use different plans.
For simple queries there won't be any difference, for complex queries probably Calcite will create better plans, but the results MUST be the same.